plugins/auth-node/api-report.md

```diff
@@ -5,12 +5,20 @@
 ```ts
 import { PluginEndpointDiscovery } from '@backstage/backend-common';
 import { Request as Request_2 } from 'express';
+import { TokenManager } from '@backstage/backend-common';
 
 // @public
-export interface BackstageIdentityResponse extends BackstageSignInResult {
-  identity: BackstageUserIdentity;
+export interface BackstageIdentityResponse<
+  I extends Identity = BackstageUserIdentity,
```

Comment:
@freben I've taken the tack of making the BackstageIdentiyResponse return BackstageUserIdentity by default and then allowing it to be used for the server/user identity for use by the IdentityApi. I am not sure if this is hair brained, so I am looking forward to hearing your comments. :D

Reply:
What I'm arguing is, since `BackstageIdentityResponse extends BackstageSignInResult`, there's no way that `IdentityApi.getIdentity` (which has nothing to do with sign-in) should return _any_ variation of `BackstageIdentityResponse`; it might have been a mistake from the start to use that return type.

To me, that signature could be changed to become its very own type that maybe, or maybe not, happens to be similar to the above.

```ts
export type IdentityApiGetIdentityResult = {
  token: string;
  identity: BackstageUserIdentity | BackstageServerIdentity;
};
```

And then reverting the change to `BackstageIdentityResponse`

Reply:
@freben I have created a new patchset with that suggestion here https://github.com/backstage/backstage/pull/13811

It needs changesets etc, but I wanted to get something in front of you to look at. Please take a look at the patch and let me know if that is what you are thinking.